CMakeLists.txt

```diff
@@ -43,418 +50,536 @@ include("cmake/ontologies.cmake")
+	src/queries/DisjunctiveBroadcaster.cpp
+	src/queries/Token.cpp
+	src/semweb/GraphSelector.cpp
+	src/queries/AnswerMerger.cpp
+	src/integration/python/utils.cpp
+	src/semweb/Triple.cpp
+	src/semweb/TripleFormat.cpp
+	src/semweb/OntologyLanguage.cpp
+	src/reasoner/prolog/semweb.cpp
+	src/integration/prolog/PrologBackend.cpp
+	src/integration/prolog/PrologTerm.cpp
+	src/semweb/SPARQLService.cpp
+	src/storage/redland/RaptorContainer.cpp
+	src/semweb/GraphRenaming.cpp
+	src/semweb/GraphRestructuring.cpp
+	src/storage/redland/RedlandModel.cpp
+	src/semweb/GraphTransformation.cpp
+	src/terms/Atom.cpp
+	src/terms/Atomic.cpp
+	src/terms/Function.cpp
+	src/terms/XSDAtomic.cpp
+	src/terms/IRIAtom.cpp
+	src/terms/Blank.cpp
+	src/terms/RDFNode.cpp
+	src/semweb/Resource.cpp
+	src/terms/Numeric.cpp
+	src/terms/String.cpp
+	src/storage/Storage.cpp
+	src/queries/QueryContext.cpp
+	src/semweb/SPARQLQuery.cpp
+	src/semweb/TripleContainer.cpp
+	src/KnowRobError.cpp
+	src/integration/python/PythonError.cpp
+	src/storage/ReifiedTriple.cpp
+	src/storage/ReifiedQuery.cpp
+	src/storage/ReificationContainer.cpp
+	src/storage/UnReificationContainer.cpp
+	src/storage/StorageInterface.cpp
+	src/semweb/GraphUnion.cpp
+	src/semweb/GraphSequence.cpp
+	src/semweb/GraphTerm.cpp
+	src/storage/mongo/MongoTriple.cpp
+	src/storage/mongo/MongoTriplePattern.cpp
+	src/storage/mongo/MongoTerm.cpp
+	src/storage/mongo/MongoTaxonomy.cpp
+	src/storage/SPARQLBackend.cpp
+	src/queries/parsers/strings.cpp
+	src/queries/parsers/terms.cpp
+	src/queries/parsers/formula.cpp
+	src/reasoner/GoalDrivenReasoner.cpp
+	src/reasoner/DataDrivenReasoner.cpp
+	src/reasoner/ReasonerEvent.cpp
+	src/semweb/TransformedOntology.cpp
+	src/reasoner/Goal.cpp
+	src/formulas/SimpleConjunction.cpp
+	src/queries/ConjunctiveQuery.cpp
+	src/semweb/RDFIndicator.cpp
+	src/integration/python/converter/shared_ptr.cpp
+	src/semweb/GraphBuiltin.cpp
+	src/reasoner/RDFGoal.cpp
+	src/reasoner/RDFGoalReasoner.cpp
+	src/reasoner/Computable.cpp
+	src/storage/Observer.cpp
+	src/storage/ObserverManager.cpp
+	src/storage/ObserverJob.cpp
+	src/queries/parsers/graph.cpp)
+
+if(NOT KNOWROB_USE_STATIC_LIB)
+	set(KNOWROB_LIBRARIES
+		${SWIPL_LIBRARIES}
+		${MONGOC_LIBRARIES}
+		${RAPTOR2_LIBRARIES}
+		${REDLAND_LIBRARIES}
+		${Python3_LIBRARIES}
+		spdlog::spdlog
+		${GTEST_LIBRARIES})
+else()
+	pkg_check_modules(RAPTOR2 REQUIRED raptor2)
+	pkg_check_modules(XSLT REQUIRED libxslt)
+	pkg_check_modules(RASQAL REQUIRED rasqal)
+	pkg_check_modules(XML2 REQUIRED libxml-2.0)
+	pkg_check_modules(SPDLOG REQUIRED spdlog)
+	pkg_check_modules(CURL REQUIRED libcurl)
+	pkg_check_modules(SSL REQUIRED libssl)
+	pkg_check_modules(CRYPTO REQUIRED libcrypto)
+	pkg_check_modules(GMP REQUIRED gmp)
+	pkg_check_modules(YAJL REQUIRED yajl)
+	pkg_check_modules(UUID REQUIRED uuid)
+	pkg_check_modules(LDAP REQUIRED ldap)
+	pkg_check_modules(LBER REQUIRED lber)
+	pkg_check_modules(PCRE REQUIRED libpcre)
+	link_directories(${MONGOC_DIR}/../../lib/
```

Comment:
why the relative path "../../lib" ? seems like a hack to me

Reply:
yes, there is a bug that always makes MONGO_LIBRARY_DIRS not the way it should be :/ I don't know why :/ it only happens with mongoc